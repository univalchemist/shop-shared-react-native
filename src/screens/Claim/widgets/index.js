export {
  default as ClaimPatientDetailBackButton,
  handlePress,
} from './ClaimPatientDetailBackButton';
export { default as ClaimReviewListItem } from './ClaimReviewListItem';
export { default as ClaimsListScreenHeader } from './ClaimsListScreenHeader';
