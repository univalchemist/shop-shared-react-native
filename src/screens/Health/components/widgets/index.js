export { default as CallToActionLink } from './CallToActionLink';
export { default as CardContainer } from './CardContainer';
export { default as CardLoader } from './CardLoader';
export { default as ErrorCard } from './ErrorCard';
export { default as TipCard } from './TipCard';
