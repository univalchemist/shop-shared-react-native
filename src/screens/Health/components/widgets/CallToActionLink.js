import React from 'react';
import { Text, Flex, Image, Divider } from '@wrappers/components';
import styled from 'styled-components/native';
import theme from '@theme';

const StyledText = styled(Text)`
  color: ${theme.colors.fonts.blackLink};
`;

const StyledImage = styled(Image)`
  width: 16px;
  height: 16px;
  align-self: center;
  margin-left: 10;
`;

export const CallToActionLink = ({ text, icon }) => {
  return (
    <Flex height={54} flexDirection="column">
      <Divider full={true} />
      <Flex
        flexDirection="row"
        paddingLeft={24}
        paddingRight={24}
        paddingTop={16}
      >
        <StyledText>{text}</StyledText>
        {icon && <StyledImage source={icon} />}
      </Flex>
    </Flex>
  );
};
