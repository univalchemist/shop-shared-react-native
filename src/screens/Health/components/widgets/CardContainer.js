import styled from 'styled-components/native';
import { TouchableHighlight } from 'react-native';
import { Box } from '@wrappers/components';

let applyStyle = comp => {
  return styled(comp)`
    background-color: ${props => props.theme.colors.white};
    height: ${props => (props.height ? props.height : 252.5)};
    border-top-color: ${props => props.color};
    border-top-width: 4;
    border-radius: 4;
    padding-top: 28;
    padding-bottom: 40;
    padding-left: 24;
    padding-right: 24;
    display: flex;
    shadow-offset: 0px 2px;
    shadow-opacity: 0.05px;
    shadow-radius: 3;
    shadow-color: #000;
  `;
};

export const CardContainer = applyStyle(Box);

export const TouchableCardContainer = applyStyle(TouchableHighlight);
