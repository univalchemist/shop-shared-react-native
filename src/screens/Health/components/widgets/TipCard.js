import React from 'react';
import { Text } from '@wrappers/components';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import styled from 'styled-components/native';
import { CallToActionLink } from './CallToActionLink';
import { TouchableHighlight, Linking, View } from 'react-native';
import { externalLinkBlack } from '@images';
import { categories, logAction } from '@store/analytics/trackingActions';

export const TouchableContainer = styled(TouchableHighlight)`
  background-color: ${props => props.theme.colors.white};
  border-radius: 4;
  padding-top: 32;
  margin-bottom: 8;
  shadow-offset: 0px 2px;
  shadow-opacity: 0.05px;
  shadow-radius: 3;
  shadow-color: #000;
  flex-grow: 1;
`;

const Topic = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  font-weight: ${props => props.theme.fontWeights.bold};
  letter-spacing: 0.3;
  padding-left: 24;
  padding-right: 24;
`;

const Source = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  letter-spacing: 0.3;
  padding-left: 24;
  padding-right: 24;
`;

const Description = styled(Text)`
  padding-top: 32;
  color: ${props => props.theme.colors.gray[8]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  letter-spacing: 0.3;
  padding-left: 24;
  padding-right: 24;
  padding-bottom: 32;
`;

export const TipCard = ({ topic, source, text, link }) => {
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableContainer
      accessible={true}
      accessibilityRole={'link'}
      underlayColor={theme.colors.touchableOverlayColor}
      onPress={() => {
        if (link) {
          Linking.openURL(link);
        }

        logAction({
          category: categories.LIFESTYLE_OVERVIEW,
          action: `Learn more ${topic}`,
        });
      }}
      theme={theme}
    >
      <>
        <View flexGrow={1}>
          <Topic accessible={true} accessibilityLabel={topic} flexGrow={1}>
            {topic}
          </Topic>
          <Source accessible={true} accessibilityLabel={source} flexGrow={1}>
            {`${intl.formatMessage({
              id: 'health.LifestyleTips.By',
            })} ${source}`.trim()}
          </Source>
          <Description accessible={true} accessibilityLabel={text} flexGrow={1}>
            {text}
          </Description>
        </View>
        <CallToActionLink
          text={intl.formatMessage({ id: 'health.learnMore' })}
          icon={externalLinkBlack}
        />
      </>
    </TouchableContainer>
  );
};
