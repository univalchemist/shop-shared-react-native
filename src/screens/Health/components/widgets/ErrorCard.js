import React from 'react';
import { TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import {
  Box,
  ErrorText,
  SectionHeadingText,
  Button,
} from '@wrappers/components';
import { LIFESTYLE_FORM } from '@routes';

export const ErrorCard = ({ color }) => {
  const navigation = useNavigation();
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableOpacity
      activeOpacity={1}
      onPress={() =>
        navigation.navigate(LIFESTYLE_FORM, { scrollToFutureMe: true })
      }
    >
      <CardContainer
        theme={theme}
        justifyContent="center"
        color={color}
        flexGrow={1}
      >
        <SectionHeadingText textAlign="center">
          {intl.formatMessage({
            id: 'health.healthScoreHistory.errorMessage.somethingWentWrong',
          })}
        </SectionHeadingText>

        <ErrorText
          mt={2}
          textAlign="center"
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.unableToShow',
          })}
          flexGrow={1}
        >
          {intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.unableToShow',
          })}
        </ErrorText>
        <ErrorText
          textAlign="center"
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.tryAgain',
          })}
          flexGrow={1}
        >
          {intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.tryAgain',
          })}
        </ErrorText>
        <Box mt={4}>
          <Button
            primary
            title={intl.formatMessage({ id: 'tryAgain' })}
            onPress={() =>
              navigation.navigate(LIFESTYLE_FORM, { scrollToFutureMe: true })
            }
          />
        </Box>
      </CardContainer>
    </TouchableOpacity>
  );
};
