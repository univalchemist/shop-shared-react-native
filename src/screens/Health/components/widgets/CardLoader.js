import React from 'react';
import { TextSkeletonPlaceholder, Box, Divider } from '@wrappers/components';
import { TouchableContainer } from './TipCard';

export const CardLoader = ({ width }) => (
  <TouchableContainer>
    <>
      <Box style={{ paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width * 0.8} />
      </Box>
      <Box style={{ paddingTop: 4, paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width * 0.5} />
      </Box>
      <Box style={{ paddingTop: 30, paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width} />
      </Box>
      <Box style={{ paddingTop: 4, paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width} />
      </Box>
      <Box
        style={{
          paddingTop: 4,
          paddingLeft: 24,
          paddingRight: 24,
          paddingBottom: 32,
        }}
      >
        <TextSkeletonPlaceholder width={width * 0.5} />
      </Box>
      <Divider full={true} />
      <Box
        style={{
          paddingTop: 16,
          paddingLeft: 24,
          paddingRight: 24,
          paddingBottom: 16,
        }}
      >
        <TextSkeletonPlaceholder width={width * 0.5} />
      </Box>
    </>
  </TouchableContainer>
);
