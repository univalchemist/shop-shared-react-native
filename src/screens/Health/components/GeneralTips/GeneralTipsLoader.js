import React from 'react';
import { Box } from '@wrappers/components';
import { CardLoader } from '@screens/Health/components/widgets/CardLoader';

export const GeneralTipsLoader = ({
  cardWidthWithSpacing,
  cardHeight,
  horizontalMarginBetweenCards,
  loaderWidthInsideCard,
}) => {
  return (
    <Box
      width={cardWidthWithSpacing}
      height={cardHeight}
      px={horizontalMarginBetweenCards / 3}
      marginTop={16}
      alignSelf="center"
    >
      <CardLoader width={loaderWidthInsideCard} />
    </Box>
  );
};
