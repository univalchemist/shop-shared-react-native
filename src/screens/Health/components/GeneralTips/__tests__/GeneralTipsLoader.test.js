import React from 'react';
import { renderForTest } from '@testUtils';
import { GeneralTipsLoader } from '../GeneralTipsLoader';
import { CardLoader } from '@screens/Health/components/widgets/CardLoader';

describe('General Tips Loader', () => {
  const generalTipsLoader = renderForTest(
    <GeneralTipsLoader
      cardWidthWithSpacing={0}
      cardHeight={252.5}
      horizontalMarginBetweenCard={3}
    />,
  );

  it('should match snapshot', () => {
    expect(generalTipsLoader.toJSON()).toMatchSnapshot();
  });

  it('should render card loader', () => {
    const cardLoader = generalTipsLoader.queryAllByType(CardLoader);

    expect(cardLoader.length).toEqual(1);
  });
});
