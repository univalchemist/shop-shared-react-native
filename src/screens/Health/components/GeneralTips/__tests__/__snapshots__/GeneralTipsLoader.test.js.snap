// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Tips Loader should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        alignSelf="center"
                        height={252.5}
                        marginTop={16}
                        px={NaN}
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "height": 252.5,
                              "marginTop": 16,
                              "paddingLeft": "NaNpx",
                              "paddingRight": "NaNpx",
                              "width": "0%",
                            },
                          ]
                        }
                        width={0}
                      >
                        <View
                          accessible={true}
                          focusable={false}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ffffff",
                                "borderRadius": 4,
                                "flexGrow": 1,
                                "marginBottom": 8,
                                "paddingTop": 32,
                                "shadowColor": "#000",
                                "shadowOffset": Object {
                                  "height": 2,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.05,
                                "shadowRadius": 3,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                },
                              ]
                            }
                          >
                            <View
                              height={22}
                              justifyContent="center"
                              style={
                                Array [
                                  Object {
                                    "height": 22,
                                    "justifyContent": "center",
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 22,
                                      "width": NaN,
                                    },
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Array [
                                      Object {
                                        "borderRadius": 4,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#f0f0f0",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "transform": Array [
                                          Object {
                                            "translateX": NaN,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          4293980400,
                                          4293125091,
                                          4293980400,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 2,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={
                                        Array [
                                          0.3,
                                          0.5,
                                          0.7,
                                        ]
                                      }
                                      startPoint={
                                        Object {
                                          "x": -1,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Object {
                                          "flex": 1,
                                          "width": NaN,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "height": 0,
                                        "width": 0,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                  "paddingTop": 4,
                                },
                              ]
                            }
                          >
                            <View
                              height={22}
                              justifyContent="center"
                              style={
                                Array [
                                  Object {
                                    "height": 22,
                                    "justifyContent": "center",
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 22,
                                      "width": NaN,
                                    },
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Array [
                                      Object {
                                        "borderRadius": 4,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#f0f0f0",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "transform": Array [
                                          Object {
                                            "translateX": NaN,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          4293980400,
                                          4293125091,
                                          4293980400,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 2,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={
                                        Array [
                                          0.3,
                                          0.5,
                                          0.7,
                                        ]
                                      }
                                      startPoint={
                                        Object {
                                          "x": -1,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Object {
                                          "flex": 1,
                                          "width": NaN,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "height": 0,
                                        "width": 0,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                  "paddingTop": 30,
                                },
                              ]
                            }
                          >
                            <View
                              height={22}
                              justifyContent="center"
                              style={
                                Array [
                                  Object {
                                    "height": 22,
                                    "justifyContent": "center",
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 22,
                                      "width": 200,
                                    },
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Array [
                                      Object {
                                        "borderRadius": 4,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#f0f0f0",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "transform": Array [
                                          Object {
                                            "translateX": -200,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          4293980400,
                                          4293125091,
                                          4293980400,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 2,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={
                                        Array [
                                          0.3,
                                          0.5,
                                          0.7,
                                        ]
                                      }
                                      startPoint={
                                        Object {
                                          "x": -1,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Object {
                                          "flex": 1,
                                          "width": 200,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "height": 0,
                                        "width": 0,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                  "paddingTop": 4,
                                },
                              ]
                            }
                          >
                            <View
                              height={22}
                              justifyContent="center"
                              style={
                                Array [
                                  Object {
                                    "height": 22,
                                    "justifyContent": "center",
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 22,
                                      "width": 200,
                                    },
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Array [
                                      Object {
                                        "borderRadius": 4,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#f0f0f0",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "transform": Array [
                                          Object {
                                            "translateX": -200,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          4293980400,
                                          4293125091,
                                          4293980400,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 2,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={
                                        Array [
                                          0.3,
                                          0.5,
                                          0.7,
                                        ]
                                      }
                                      startPoint={
                                        Object {
                                          "x": -1,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Object {
                                          "flex": 1,
                                          "width": 200,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "height": 0,
                                        "width": 0,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "paddingBottom": 32,
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                  "paddingTop": 4,
                                },
                              ]
                            }
                          >
                            <View
                              height={22}
                              justifyContent="center"
                              style={
                                Array [
                                  Object {
                                    "height": 22,
                                    "justifyContent": "center",
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 22,
                                      "width": NaN,
                                    },
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Array [
                                      Object {
                                        "borderRadius": 4,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#f0f0f0",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "transform": Array [
                                          Object {
                                            "translateX": NaN,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          4293980400,
                                          4293125091,
                                          4293980400,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 2,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={
                                        Array [
                                          0.3,
                                          0.5,
                                          0.7,
                                        ]
                                      }
                                      startPoint={
                                        Object {
                                          "x": -1,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Object {
                                          "flex": 1,
                                          "width": NaN,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "height": 0,
                                        "width": 0,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            backgroundColor="gray.5"
                            height={1}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#e3e3e3",
                                  "height": 1,
                                  "width": "100%",
                                },
                              ]
                            }
                            width="100%"
                          />
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "paddingBottom": 16,
                                  "paddingLeft": 24,
                                  "paddingRight": 24,
                                  "paddingTop": 16,
                                },
                              ]
                            }
                          >
                            <View
                              height={22}
                              justifyContent="center"
                              style={
                                Array [
                                  Object {
                                    "height": 22,
                                    "justifyContent": "center",
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 22,
                                      "width": NaN,
                                    },
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Array [
                                      Object {
                                        "borderRadius": 4,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#f0f0f0",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "transform": Array [
                                          Object {
                                            "translateX": NaN,
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          4293980400,
                                          4293125091,
                                          4293980400,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 2,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={
                                        Array [
                                          0.3,
                                          0.5,
                                          0.7,
                                        ]
                                      }
                                      startPoint={
                                        Object {
                                          "x": -1,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Object {
                                          "flex": 1,
                                          "width": NaN,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "height": 0,
                                        "width": 0,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
